hyperspy/misc/eds/utils.py

```diff
@@ -652,25 +651,21 @@ def get_abs_corr_cross_section(composition, number_of_atoms, take_off_angle,
 
     number_of_atoms = stack(number_of_atoms, show_progressbar=False).data
 
-    #calculate the total_mass per pixel, or mass thicknessself.
+    #calculate the total_mass in kg/m^2, or mass thickness.
     total_mass = np.zeros_like(number_of_atoms[0], dtype = 'float')
     for i, (weight) in enumerate(atomic_weights):
-        total_mass += (number_of_atoms[i] * weight / Av / probe_area / 1E-15)
-
-    # determine mass absorption coefficients and convert from cm^2/g to m^2/atom.
+        total_mass += (number_of_atoms[i] * weight / Av / 1E3 / probe_area / 1E-18)
+    # determine mass absorption coefficients and convert from cm^2/g to m^2/kg.
     to_stack = material.mass_absorption_mixture(
         weight_percent=material.atomic_to_weight(composition)
         )
     mac = stack(to_stack, show_progressbar=False) * 0.1
-
     acf = np.zeros_like(number_of_atoms)
-    constant = 1/(Av * math.sin(toa_rad) * probe_area * 1E-16)
-
-    #determine an absorption coeficcient per element per pixel.
+    csc_toa = 1/math.sin(toa_rad)
```

Comment:
This looks like a non-negligible change in the calculation, but it doesn't affect the test results. Is it expected?

Reply:
So this made the estimate sample thickness so small that there was no absorption. The zeta error made it so that there was only one loop in the zeta correction. Therefore it was only serendipity that they ended up matching. When I fixed the zeta error the corresponding one in the cross sections function appeared. No I agree it's non-negligible, just luck that they matched before this.

Reply:
Basically the probe_area was being incorporate twice and it shouldn't have been.